packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts

```diff
@@ -221,11 +222,35 @@ export class Messenger {
                             if (tool.type === ToolType.FsWrite) {
                                 session.setShowDiffOnFileWrite(true)
                             }
+                            if (
```

Comment:
I know not specific to your PR but this function is way too long and difficult to read, can we take this part out into a helper function?

Reply:
Agreed, I can follow up on this part later